src/components/Table/TableCell/TableCell.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import clsx from 'clsx';
+
+import {TableCellProps} from './TableCell.types';
+import './TableCell.scss';
+import {Typography} from '~/components';
+import {capitalize} from '~/utils/helpers';
+
+export const TableCell: React.FC<TableCellProps> = ({
+    className,
+    component = 'td',
+    horizontalAlign = 'left',
+    verticalAlign = 'center',
+    children,
+    ...props
+}) => {
+
+    // what's the correct naming convention for these cell scss classes?
+    return React.createElement(
+        component,
+        {
+            className: clsx(
+                'moonstone-table_cell',
+                'horizontalAlign' + capitalize(horizontalAlign),
+                'verticalAlign' + capitalize(verticalAlign),
+                className
+            ),
+            ...props
+        },
+        component === 'th'
```

Comment:
I'm wondering if we should split it into two components `TableCellHead` and `TableCellBody`. the `th`  element has many differences (typography, sorting, no hover/selection ....) whereas the `td` element is just a wrapper. What do you think ?

Reply:
Hmm yes, that could make the components simpler. I'm going to make `TableHeadCell` and `TableBodyCell`.